server/channels/utils/license_test.go

```diff
@@ -67,7 +67,8 @@ func TestValidateLicense(t *testing.T) {
 	})
 
 	t.Run("should reject invalid license in test service environment", func(t *testing.T) {
-		os.Setenv("MM_SERVICEENVIRONMENT", model.ServiceEnvironmentTest)
+		err := os.Setenv("MM_SERVICEENVIRONMENT", model.ServiceEnvironmentTest)
+		require.NoError(t, err, "failed to set environment variable")
 		defer os.Unsetenv("MM_SERVICEENVIRONMENT")
```

Comment:
Nit pick: You can also use `T.Setenv` instead, as that also cleans up after itself. Not a blocker for the PR.